### ISSUE TITLE
AOI Selection via Predefined Bounds

### ISSUE THREAD

**Issue by aboydnw** [NASA-IMPACT/veda-ui#212]
## Description
As a basic scientist
I want to select an AOI using predefined bounds
So that I can more quickly select the area that I am interested in

Part of the Analysis page: https://www.figma.com/proto/SA0u0sVdw5vM7g9ulV5Xgw/Wireframes?page-id=1498%3A2972&node-id=1601%3A3246&viewport=-615%2C134%2C0.52&scaling=min-zoom

This ticket came to be based on some feedback from Alexey when reviewing wire of the analysis page. Essentially, many of their users will have predefined bounds that they care about (state, county, fire district, etc.) 

TBD on exactly which predefined bounds we will want to include in our first version.

## Acceptance Criteria
- [ ] Given I have clicked the Edit button on the Analysis page, when the sidebar opens up then I see pre-defined bounds 
- [ ] Given I have clicked the Edit button on the Analysis page, when I select a pre-defined AOI then the map and coordinates respond to reflect that selection

**Comment by aboydnw** [comment]
Related to, but not dependent on #205

**Comment by aboydnw** [comment]
Questions:
1. What sorts of pre-defined options would be most useful for scientists?
2. Are these options specific to the datasets they want to view?
3. Are these options specific to the region / continent that they are viewing?
4. Will scientists want to define their own regions to be referenced later?

**Comment by ashiklom** [comment]
> What sorts of pre-defined options would be most useful for scientists?

The obvious one is standard political geographic units — e.g., countries, states, counties.

Beyond that, there are discipline- or application-specific ones. For example, an important unit for fires is [NIFC Predictive Service Areas](https://data-nifc.opendata.arcgis.com/maps/e580f3fd4d644366b121676714d69c2d), and an important one for air quality is [Air Quality Management Districts](https://ww2.arb.ca.gov/california-map-local-air-district-websites).

> Are these options specific to the datasets they want to view?

I think analysis context is more important than dataset — i.e., if you're analyzing fires, you might want to be able to look at as many datasets as possible, including ones that aren't _strictly_ about fires but may provide relevant context or downstream impacts (e.g., precipitation, weather, soil moisture, land cover). Not sure how to structure this from a UI standpoint without overwhelming users — maybe, have a "basic" set of units that are fairly common and universal (like political boundaries), and then an "advanced" menu that provides a longer searchable list of these.

> Are these options specific to the region / continent that they are viewing?

Usually, yes. US States aren't relevant if you're looking at Africa; etc. Many of the more advanced divisions (e.g., NIFC PSAs, air quality management districts) will be limited to the USA or even specific regions or states.

> Will scientists want to define their own regions to be referenced later?

Custom polygons defined on the fly using a simple "draw a polygon" interface or standard syntax (e.g., GeoJSON uploaded or pasted into a box; see [AppEEARs for an example interface](https://appeears.earthdatacloud.nasa.gov/task/area)) would be useful. Persistently storing these online so they can be "referenced later" is probably a lower priority feature, unless it's trivial to implement.

@dfelikson @eorland Thoughts?

**Comment by dfelikson** [comment]
Building on @ashiklom's answers ...

> What sorts of pre-defined options would be most useful for scientists?

You could maybe add continents (N. America, Asia, etc.) and oceans (Pacific, Atlantic, etc.) to Alexey's list.

> Will scientists want to define their own regions to be referenced later?

An alternative option to persisting these online is giving users the ability to download the AOI and then re-upload it the next time they use the system.

**Comment by aboydnw** [comment]
Really helpful, thank you @ashiklom and @dfelikson ! One last question to followup with:
### Given that there is so many potential presets to provide to users, in each of your respective disciplines, what would be the most useful type of preset?
Some potential types that I see referenced above:
- US state, county, or municipality
- Country/national boundaries
- Continent and/or ocean
- Discipline-specific boundaries such as [NIFC Predictive Service Areas](https://data-nifc.opendata.arcgis.com/maps/e580f3fd4d644366b121676714d69c2d) or [Air Quality Management Districts](https://ww2.arb.ca.gov/california-map-local-air-district-websites)


## DS Team
@faustoperez @ricardoduplos @nerik a few design challenges I see based on their answers above. We can address in these ticket or if you all want to propose a first "chunk" to build and get feedback on, I would appreciate those thoughts as well.

1. Allowing users to easily select preset values that make sense based on the geographic region they are interested in (aka states in US vs provinces in Canada)
2. Similar to above, not being flooded with a list of all potential political boundaries or other presets (aka if we gave every state or even every country in the world as a preset)
3. (Potentially) Allowing users to select some presets that are generic to all datasets and analyses (political boundaries like states or counties)
4. The easiest way for users to use any custom-defined polygons later on (maybe this is a separate ticket)
5. Does the current analysis page facilitate analysis-specific workflows? Keying off of Alexey's comment above. This is more a blue sky question and probably not something we can directly address on this ticket.

**Comment by dfelikson** [comment]
My 2 cents ... I would list all regions available for all datasets. Most (or probably all) of the thematic areas have global relevance and, even though some datasets might be spatially confined (e.g., the land cover dataset that we have in the coastal risk Discovery), the other datasets within that thematic area are global.

**Comment by nerik** [comment]
> 4. The easiest way for users to use any custom-defined polygons later on (maybe this is a separate ticket)

Support for this (custom geojson/shapefile upload) has just been merged https://github.com/NASA-IMPACT/veda-ui/pull/381

**Comment by nerik** [comment]
Split into:
https://github.com/NASA-IMPACT/veda-ui/issues/404 AoI Selection via Custom-defined polygons
https://github.com/NASA-IMPACT/veda-ui/issues/405 AoI via predefined generic boundaries
https://github.com/NASA-IMPACT/veda-ui/issues/406 AoIs for Discipline-specific boundaries

cc @aboydnw
